### ISSUE TITLE
AttributeError: 'types.SimpleNamespace' object has no attribute 'get_registered_object' from return keras.saving.get_registered_object(config["registered_name"]) in keras_nlp\src\utils\preset_utils.py

### ISSUE THREAD

**Issue by JingsaiLu** [keras-team/keras#19456]
want to run the demo in the local by following the [getting started guide ](https://www.kaggle.com/code/nilaychauhan/get-started-with-gemma-using-kerasnlp?utm_source=%E3%80%90%E6%B4%BB%E5%8A%A8%E7%A1%AE%E8%AE%A4%E3%80%91%E6%82%A8%E5%B7%B2%E6%88%90%E5%8A%9F%E6%8A%A5%E5%90%8D%E5%8F%82%E5%8A%A0+GDG+B...+%28%E7%BE%A4%E5%8F%91%E9%82%AE%E4%BB%B6%E5%88%9B%E5%BB%BA%E6%97%B6%E9%97%B4%EF%BC%9A2024-04-05%29&utm_medium=Email)
and config the kaggle API credentials with[ this doc](https://github.com/Kaggle/kaggle-api/blob/main/docs/README.md#api-credentials) 
but get error when run the setup 5:
`import keras
import keras_nlp
import os

os.environ["KERAS_BACKEND"] = "jax"

gemma_lm = keras_nlp.models.GemmaCausalLM.from_preset("gemma_2b_en")`

![image](https://github.com/keras-team/keras/assets/3600912/d55131da-bb8c-4e8f-9a62-653f8e5ac18f)

what is the reason for this ? and how to fix that ?

**Comment by SuryanarayanaY** [comment]
Hi @jsplyy ,

It seems to be version issue for me. Could you please confirm the keras_nlp and keras version being used ? Please install latest versions of both and let us know the outcome. Thanks!

**Comment by jsplyy** [comment]
Hi @SuryanarayanaY 
I am using, 
Name: keras-nlp
**Version: 0.9.1**
Name: keras
**Version: 2.10.0**

try to update keras with command: pip upgrade keras
it will be updated to 3.1.1, 
![image](https://github.com/keras-team/keras/assets/3600912/b05e67f5-7433-44ad-8f5f-2a5931c7e3d7)

but when try to update keras_nlp with command: pip install --upgrade keras_nlp
keras-2.10.0 will be installed again.
**keras_nlp seems to depends on and install 2.10.0 ?**
![image](https://github.com/keras-team/keras/assets/3600912/6c0a8bb8-7cf3-486f-9607-f124967e1b33)

I checked the keras-nlp doc https://github.com/keras-team/keras-nlp/tree/master
it documented that both keras2 and keras3 are supported.
![image](https://github.com/keras-team/keras/assets/3600912/d8e9017e-391b-4db8-93f2-d55df9e3e984)

**Comment by SuryanarayanaY** [comment]
> keras_nlp seems to depends on and install 2.10.0 keras?

Hi @jsplyy ,

No, it don't have any dependency on Keras 2.10.0 . Please ensure all the dependencies in requirements.txt installed. Keras_nlp needs Tensorflow also. May be you might have TF2.10v which might be reason for this behaviour.

**Comment by dongex** [comment]
I also faced the same error and I have installed Tensorflow

**Comment by jvrapp** [comment]
@gustheman I'm still having this issue. When installing keras_nlp==0.12.1 it installs keras==2.10.0 and tf 2.10.1. Also when using keras_nlp==0.14.1 it doesn't let to import to jupyter due missing ops module

**Comment by mattdangerw** [comment]
KerasNLP no longer supports Keras 2, we are focusing on Keras 3. This means a few things...

* The latest versions of KerasNLP or Keras working with TensorFlow 2.14 or earlier.
* For TensorFlow 2.15. You need the weird install line @jvrapp showed. This is because Tensorflow 2.15 will pull in Keras 2 during installation.
```
!pip install -U -q keras-nlp
!pip install -q -U keras>=3
```
 * For TensorFlow 2.16 and greater. Everything should work normally. Tensorflow 2.16 will pull in Keras 3. Everything cooperates.

So TL;DR, make sure you base installation has tensorflow >2.16 and tensorflow-text >2.16 and things should be fine.

**Comment by mattdangerw** [comment]
In a colab environment, this should be quite simple. Colab now comes with TensorFlow 2.17. So

`!pip install keras-nlp` will do the right thing and pull in all necessary dependancies.

With a base environment where Tensorflow is older, you might want to do something like this to make sure your tf is upgraded...

`!pip install -U keras-nlp tensorflow tensorflow-text`

Hope that helps!

**Comment by Gopi-Uppari** [comment]
Hi @JingsaiLu,

Could you please confirm if this issue is resolved for you with the above comment ? Please feel free to close the issue if it is resolved ?

Thank you.
